### PR TITLE
Khoury auth flow

### PR DESCRIPTION
**Pull request by NilsBacke**

### CODE REVIEW

**Comment by NilsBacke** [comment]
[Log in through Khoury SSO](https://trello.com/c/bJyRBBEu/193-log-in-through-khoury-sso)

File: api/app/models/user.rb [hunk 1]
```diff
@@ -2,36 +2,27 @@
...
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  nu_id           :string
 #
 # Indexes
 #
-#  index_users_on_email                 (email) UNIQUE
-#  index_users_on_reset_password_token  (reset_password_token) UNIQUE
-#  index_users_on_username              (username)
+#  index_users_on_email  (email) UNIQUE
```
Comment (ArunJ98) [inline api/app/models/user.rb:20]: Can we also index on username - Will be useful to get back list of sorted students.

File: package.json [hunk 2]
```diff
@@ -14,6 +14,8 @@
   "scripts": {
     "start": "cd infrastructure/develop && ./compose-build.sh && ./run-servers.sh dev-build",
     "prod-start": "cd infrastructure/develop && ./compose-build.sh && ./run-servers.sh prod-build",
+    "no-docker-start:frontend": "./no_docker_frontend_start.sh",
+    "no-docker-start:backend": "./no_docker_backend_start.sh",
```
Comment (NilsBacke) [inline package.json:18]: I just got the new macbook pro with the M1 chip so it doesn't support docker. So we can use this for people who can't/don't want to use docker and just use rails with postgres natively.

File: api/app/models/user.rb (hunk 1 above)
Comment (NilsBacke) [reply api/app/models/user.rb:20]: The issue with that is we set the username as first_name + last_name, and Ali said that are a bunch of duplicate names in Khoury.

File: api/config/environments/production.rb [hunk 3]
```diff
@@ -1,5 +1,7 @@
 Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
+  ENV['ROOT_URL'] = "http://graduatenu.com"
+  ENV['FRONTEND_URL'] = "http://graduatenu.com"
```
Comment (NilsBacke) [inline api/config/environments/production.rb:4]: add `www` here

File: frontend/src/App.tsx [hunk 4]
```diff
@@ -31,9 +30,8 @@ export const App = ({
         <Router>
           <Switch>
             <Route path="/home" component={HomeWrapper} />
+            <Route path="/redirect" component={RedirectScreen} />
             <Route path="/onboarding" component={OnboardingInfoScreen} />
```
Comment (NilsBacke) [inline frontend/src/App.tsx:34]: Add `/advisor` path redirect to notifications page

File: frontend/src/Onboarding/Onboarding.tsx [hunk 5]
```diff
@@ -169,27 +188,32 @@ class OnboardingComponent extends React.Component<OnboardingProps> {
     );
   }
 
-  componentWillMount() {
-    // make an API request to searchNEU to get the supported majors and their corresponding plans.
-    this.props.fetchMajorsAndPlans();
-  }
-
   render() {
     // fullName will be an empty string if this is the user's first time visiting the site
```
Comment (NilsBacke) [inline frontend/src/Onboarding/Onboarding.tsx:192]: remove comment

File: frontend/src/Onboarding/RedirectScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      });
+      cookies.set("auth_token", cookie, {
+        path: "/",
+        domain: window.location.hostname,
+      }); // set persisting cookie
+      if (cookie) {
+        fetchUser(cookie).then(response => {
+          dispatch(setFullNameAction(response.user.username));
+          const maj = findMajorFromName(response.user.major, majors);
+          console.log(response.user.major);
```
Comment (ArunJ98) [inline frontend/src/Onboarding/RedirectScreen.tsx:52]: can we get rid of these console logs?

**Review by ArunJ98 — APPROVED** [review]
Looks good, might need to change the seed task you created to include nu-id/get rid of password?

**Pull request merged**